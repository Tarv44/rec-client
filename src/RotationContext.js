import React from 'react';

const RotationContext = React.createContext({
    updateUser: () => {}
})

export default RotationContext;