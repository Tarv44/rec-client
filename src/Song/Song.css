.song-section {
    border-bottom: solid 3px black;
    background-color: white;
}

.song-title {
    font-size: 24px;
    margin: 0;
    color: #4C00F0;
}

.song-added,
.song-link {
    margin: 2px 0 0 0;
    font-style: italic;
    font-weight: normal;
    text-align: right;
}

.song-added {
    font-size: 12px;
}

.song-link {
    color: #454545;
    font-size: 13px;
}

.song-div {
    margin: 0 5%;
    padding: 2% 0;
}

.song-header {
    text-align: right;
    min-width: 60%;
    padding-bottom: 0;
}

.song-details {
    border-bottom: solid black 1px;
    padding-top: 0;
}

.song-details > * {
    margin-bottom: .5em;
    margin-top: 0;
}

.song-comments > p {
    background-color: white;
    color: #454545;
    display: block;
}

.song-comments * {
    margin: .2em 0;
}

.username {
    /* text-decoration: #0CF0AB underline; */
    font-weight: bold;
    color: #4C00F0;
}