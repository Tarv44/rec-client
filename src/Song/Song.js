import { Component } from 'react';
import RotationContext from '../RotationContext';
import AddComment from '../AddComment/AddComment';
import moment from 'moment';
import './Song.css';

export default class Song extends Component {
    static contextType = RotationContext
    render() {
        const song = this.props.song
        const added_by = this.props.users.find(user => user.id === song.added_by).username
        const comments = song.comments
            ? song.comments.map((comm, i) => {
                const username  = this.props.users.find(user => user.id === comm.created_by).username
                return <p key={i}><span className='username'>{username}:</span> {comm.message}</p>
            })
            : null

        const newComment = this.context.current_user.username 
            ? <AddComment 
                handleComment={this.props.handleComment} 
                handleCommentSubmit={this.props.handleCommentSubmit} 
                songIdx={this.props.index}
            />
            : <p>Log in to leave a comment</p>
        return (
            <div className='song-section'>
                <div className="song-header">
                    <h3 className='song-title'>{song.title}</h3>
                    <h4 className='song-added'>{moment(song.date_added).format('l')} - <span className='username'>{added_by}</span></h4>
                    {song.url_link ? <a href={song.url_link} target='_blank' rel="noreferrer" className='song-link'>Track Link</a> : null}
                </div>
                
                <div className='song-details'>
                    <p><b>Artist:</b> {song.artist}</p>
                    {song.album ? <p><b>Album:</b> {song.album}</p> : null}
                </div>
                
                <div className='song-comments'>
                    {comments}
                    {newComment}
                </div>
                
            </div>
        )
    }
}