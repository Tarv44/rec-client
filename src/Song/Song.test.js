import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import Song from './Song'
import store from '../dummy-store'
import RotationContext from '../RotationContext'


describe('----- Song -----', () => {
    const context = {
        current_user: {
            id: 1,
            username: 'user'
        },
        current_exchanges: store.exchanges
    }
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render((
            <RotationContext.Provider value={context}>
                <Song users={store.users} song={store.songs[0]}/>
            </RotationContext.Provider>   
        ), div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <RotationContext.Provider value={context}>
                    <Song users={store.users} song={store.songs[0]}/>
                </RotationContext.Provider>    
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})