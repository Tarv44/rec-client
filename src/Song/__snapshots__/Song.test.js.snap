// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- Song ----- renders the UI as expected 1`] = `
<div
  className="song-section"
>
  <div
    className="song-header song-div"
  >
    <h3
      className="song-title"
    >
      Do you want more?!?!
    </h3>
    <h4
      className="song-added"
    >
      2/18/2021
       - 
      <span
        className="username"
      >
        user
      </span>
    </h4>
  </div>
  <div
    className="song-details song-div"
  >
    <p>
      <b>
        Artist:
      </b>
       
      The Roots
    </p>
    <p>
      <b>
        Album:
      </b>
       
      Do you want more?!?!
    </p>
  </div>
  <div
    className="song-comments song-div"
  >
    <form
      autoComplete="off"
      className="song-form"
      onSubmit={[Function]}
    >
      <input
        className="comment-input"
        maxLength="1500"
        name="comment"
        onChange={[Function]}
        type="text"
      />
      <button
        className="comment-button"
        type="submit"
      >
        Comment
      </button>
    </form>
  </div>
</div>
`;
